### PR TITLE
VACMS-16191 Add test coverage for Resources & Support app

### PR DESCRIPTION
**Pull request by randimays**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
The Resources & Support app had some opportunities for unit test coverage and a Cypress suite.

### Unit tests
Notes:
- I moved a bunch of functions from within a `useEffect` to the outer scope so they could be unit tested easily. The functionality of the application should not have changed at all.
- We are limited a lot in unit tests by our inability to interact with web components and the shadow DOM. This is largely mitigated by the Cypress test suites

**Before**
284718072-5f8de440-6b6a-41e7-a820-7465cc82373c

**After**
Screenshot 2024-11-20 at 2 09 03 PM

### Cypress
Our Cypress suite has:
- A suite for both desktop and mobile (separately)
- Validation for the inputs including open/closed search area on mobile
- Checking correct text and URLs for results
- Validation for correct pagination
- Checking correct display for search terms with no results
- Checking a sitewide search (as well as the default R&S search)

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16191

## Testing done
Tested the app manually locally and ran all unit and Cypress tests. No functionality should have changed so I didn't add screenshots here.

## What areas of the site does it impact?
`/resources` and `/resources/search`

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] Linting warnings have been addressed

### Error Handling

- [x] Browser console contains no warnings or errors.

### CODE REVIEW

File: src/applications/resources-and-support/components/SearchResultList.jsx [hunk 1]
```diff
@@ -13,10 +11,7 @@ export default function SearchResultList({
 }) {
   return (
     // eslint-disable-next-line jsx-a11y/no-redundant-roles
-    <ul
-      className="usa-unstyled-list vads-u-padding-x--1 desktop-lg:vads-u-padding-x--0"
-      role="list"
```
Comment (randimays) [inline src/applications/resources-and-support/components/SearchResultList.jsx:18]: Laura mentioned in DMs in an earlier that we did not need `role="list"` on `<ul>` elements.

File: src/applications/resources-and-support/content-modeling.js [hunk 2]
```diff
@@ -1,51 +0,0 @@
-/**
```
Comment (randimays) [inline src/applications/resources-and-support/content-modeling.js:1]: It wasn't clear to me what `content-modeling` meant for this app, and this was just a constants mapping anyway, so I moved it to `constants`.

File: src/applications/resources-and-support/hooks/useGetSearchResults.js [hunk 3]
```diff
@@ -23,8 +21,141 @@ const countKeywordsFound = (searchableString = '', keyword = '') => {
   return keywordsFound;
 };
 
-const countWholePhraseMatchesFound = (searchableString = '', keyword = '') =>
-  searchableString.toLowerCase()?.split(keyword.toLowerCase())?.length - 1;
+export const countWholePhraseMatchesFound = (
```
Comment (randimays) [inline src/applications/resources-and-support/hooks/useGetSearchResults.js:24]: Looks like a lot of changes here, but I just made a bunch of utility functions out of the code that was in the `useEffect` so I could access it for unit testing.

File: src/applications/resources-and-support/tests/components/articles.json [hunk 4]
```diff
@@ -1,114 +0,0 @@
-[
```
Comment (randimays) [inline src/applications/resources-and-support/tests/components/articles.json:1]: This became `/tests/articles.json` and I replaced the test data with real data.

**Comment by jilladams** [comment]
@randimays this is the only open Sitewide vets-website PR I see right now. Once you move out of Draft, can you confirm if [Curt's notes in devops channel](https://dsva.slack.com/archives/C04CYC4LMU6/p1732131652244929?thread_ts=1732129280.235259&cid=C04CYC4LMU6) are correct that you are able to merge with just a team approval, and not Platform approval?

**Review by chriskim2311 — APPROVED** [review]
LGTM! Great work, tests look good and are passing.

**Comment by randimays** [comment]
@jilladams Confirmed that I was able to merge this with only Chris's approval because the changes were siloed to the `src/applications` folder. A lot of our work crosses over into the `src/platform` folder so we have to wait on Platform when that happens ([example](https://github.com/department-of-veterans-affairs/vets-website/pull/33147)).

**Pull request merged**